experimental/dds/tree2/src/util/utils.ts

```diff
@@ -32,6 +32,13 @@ export type RecursiveReadonly<T> = {
  */
 export type Mutable<T> = { -readonly [P in keyof T]: T[P] };
 
+/**
+ * Make all field required and omits fields whose ony valid value would be `undefined`.
```

Comment:
Might be worth adding a note here that this is analogous to `Required<T>` except it tolerates 'optional undefined'.